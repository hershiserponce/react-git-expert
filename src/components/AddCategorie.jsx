import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategorie = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCategorie(categories=>[inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };
  return (
    <>
      <form onSubmit={onSubmit} aria-label='form'>
        <input
          type="text"
          placeholder="Buscar Gif"
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </>
  );
};



AddCategorie.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}