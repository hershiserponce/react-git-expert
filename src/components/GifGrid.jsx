
import { GifItem } from "./index";
import { useFetchGif } from "../hooks/useFetchGif";
import PropTypes from "prop-types"

export const GifGrid = ({ category }) => {

  const {gifData, isLoading} = useFetchGif(category);
 

  return (
    <>
    <h3>{category}</h3>

    {
      // isLoading ? (<h2>cargando...</h2>) : null}
      isLoading && (<h2>cargando...</h2>) }

        <div className="card-grid">
      {gifData.map((image) => (
        <GifItem key={image.id} {...image} /> 
        
          ))}
          </div>
    </>
  );
};


GifGrid.propTypes= {
  category: PropTypes.string.isRequired,
}