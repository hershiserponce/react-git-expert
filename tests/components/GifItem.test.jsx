
import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('pruebas en <GifItem/>', () => { 

    const title = 'pepe'
    const url = 'www.google.com.ni'

    test(' debe de hacer match con el snapshop', () => {
        const { container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     });

     test('debe de mostrar la imagen con el url de prueba y el alt indicado', () => { 
        render(<GifItem title={title} url={url}/>)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
      })

      test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
       })

 })