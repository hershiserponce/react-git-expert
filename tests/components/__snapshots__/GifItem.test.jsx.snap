// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem/>  debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pepe"
      src="www.google.com.ni"
    />
    <p>
      pepe
    </p>
  </div>
</div>
`;
